Vector.c

```diff
@@ -242,6 +255,56 @@ Object* Vector_remove(Vector* this, int idx) {
    }
 }
 
+Object* Vector_softRemove(Vector* this, int idx) {
+   assert(idx >= 0 && idx < this->items);
+
+   Object* removed = this->array[idx];
+   this->array[idx] = NULL;
+
+   this->dirty_count++;
+   if (this->dirty_index < 0 || idx < this->dirty_index) {
+      this->dirty_index = idx;
+   }
+
+   if (this->owner) {
+      Object_delete(removed);
+      return NULL;
+   }
+   return removed;
```

Comment:
```suggestion

   return removed;
```

Just add a blank here; better visual grouping.